bitnami/postgresql-ha/templates/pgpool/deployment.yaml

```
@@ -227,7 +227,7 @@ spec:
               valueFrom:
                 secretKeyRef:
                   name: {{ include "postgresql-ha.postgresqlSecretName" . }}
-                  key: password
+                  key: postgres-password
```

Password secret could not contain `postgres-password` key if `postgresql.username == 'postgres'`:  https://github.com/bitnami/charts/blob/375ee44a7787e8dde6573924fb87994a83e64d2f/bitnami/postgresql-ha/templates/postgresql/secrets.yaml#L18-L24

---

Thanks for reviewing the code. Unfortunately, I found no documentation about postgresqlUsername at 
https://github.com/bitnami/charts/tree/main/bitnami/postgresql
or
https://github.com/bitnami/charts/tree/main/bitnami/postgresql-ha

There is just a piece of information at
https://github.com/bitnami/charts/tree/main/bitnami/postgresql#automated-update-using-a-password-update-job
, but about "auth.postgresPassword".

See https://github.com/bitnami/charts/tree/main/bitnami/postgresql#global-parameters as well.
No information about what you're mentioning here.

So this is other (documentation)  bug?

---

No and sorry if you thought that. We appreciate any help and any contribution.

Just apply this suggestion, give it a try and check if it fits to you:

```suggestion
                  {{- if not (eq (include "postgresql-ha.postgresqlUsername" .) "postgres") }} 
                  key: postgres-password
                  {{- else }}
                  key: password
                  {{- end }}
```